pkg/broker/broker.go

```diff
@@ -619,6 +621,16 @@ func (a AnsibleBroker) Provision(instanceUUID uuid.UUID, req *ProvisionRequest,
 	//
 	// if err is not nil, we will just bubble that up
 
+	alreadyInProgress, err := a.isProvisionInProgress(serviceInstance)
+	if err != nil {
+		return nil, fmt.Errorf("An error occurred while trying to determine if a provision job is already in progress for instance: %s", serviceInstance.ID)
+	}
+
+	if alreadyInProgress {
+		a.log.Infof("Provision requested for instance %s, but job is already in progress", serviceInstance.ID)
+		return nil, ErrorProvisionInProgress
```

Comment:
I would change `nil` to `&ProvisionResponse{Operation: token}` somehow we'll need the token not sure yet.

Reply:
:+1: deleted my comment that was essentially the same as this.